### PR TITLE
Codeanalysis analyzers rulesets

### PR DESCRIPTION
**Pull request by COsborn2**
Warnings intentionally left

CA2007 - Call ConfigureAwait on awaited task
CA2008 - Do not create tasks without passing a TaskScheduler
CA2208 - Method passes parameter name as message argument to ArgumentException constructor. Replace this argument with a descriptive message.

CA1816 - VirtualMemoryPtr.ReleaseHandle() calls GC.SuppressFinalize(object), a method that is typically only called within an implementation of IDisposable.Dispose. Refer to IDisposable pattern

### CODE REVIEW

File: src/Chapter07/Listing07.18.DefiningAnAbstractClass.cs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ﻿namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter07.Listing07_18
 {
-// With a concrete implementation we could use our PdaItem object
+    // With a concrete implementation we could use our PdaItem object
```
Comment (COsborn2) [inline src/Chapter07/Listing07.18.DefiningAnAbstractClass.cs]:
```suggestion
// With a concrete implementation we could use our PdaItem object
```

File: src/Chapter08/Listing08.04.CallingExplicitInterfaceMemberImplementations.cs [hunk 2]
```diff
@@ -2,8 +2,10 @@
 {
     using System;
     using Listing08_02;
-// In a production implementation we would do something with our declared Contacts
+    // In a production implementation we would do something with our declared Contacts
```
Comment (COsborn2) [inline src/Chapter08/Listing08.04.CallingExplicitInterfaceMemberImplementations.cs]:
```suggestion
// In a production implementation we would do something with our declared Contacts
```

File: src/Chapter08/Listing08.05.ExplicitInterfaceImplementation.cs [hunk 3]
```diff
@@ -2,8 +2,10 @@
 {
     using System;
     using Listing08_02;
-// In a production implementation we would do something with our declared Contacts
+    // In a production implementation we would do something with our declared Contacts
```
Comment (COsborn2) [inline src/Chapter08/Listing08.05.ExplicitInterfaceImplementation.cs]:
```suggestion
// In a production implementation we would do something with our declared Contacts
```

File: src/Chapter14/Listing14.18.CSharpConceptualEquivalentOfEventCILCode.cs [hunk 4]
```diff
@@ -1,8 +1,9 @@
 ﻿namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter14.Listing14_18
 {
     using System;
-// In an actual implementation _OnTemperatureChange would be used
+    // In an actual implementation _OnTemperatureChange would be used
```
Comment (COsborn2) [inline src/Chapter14/Listing14.18.CSharpConceptualEquivalentOfEventCILCode.cs]:
```suggestion
// In an actual implementation _OnTemperatureChange would be used
```

File: src/Chapter18/Listing18.08.DecoratingAPropertyWithAnAttribute.cs [hunk 5]
```diff
@@ -2,6 +2,8 @@ namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter18.Listing18_08
 {
     using System;
 
+#pragma warning disable CA1019 //
```
Comment (COsborn2) [inline src/Chapter18/Listing18.08.DecoratingAPropertyWithAnAttribute.cs]:
```suggestion
#pragma warning disable CA1019 // Attributes are not full implemented in this listing
```

File: src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs [hunk 6]
```diff
@@ -24,7 +25,7 @@ public string LastName
                 if(value == null)
                 {
                     // Report error
-                    throw new ArgumentNullException(nameof(value));
+                    throw new ArgumentNullException("value");
```
Comment (Keboo) [inline src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs:28]: Why change away from nameof?

File: src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs [hunk 7]
```diff
@@ -34,7 +35,7 @@ public string LastName
                     if(value == "")
                     {
                         throw new ArgumentException(
-                            "LastName cannot be blank.", nameof(value));
+                            "LastName cannot be blank.", "value");
```
Comment (Keboo) [inline src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs:38]: Why change away from nameof?

**Review by Keboo — APPROVED** [review]

File: src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs (hunk 6 above)
Comment (COsborn2) [reply src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs:28]: Updated to match book's text. I'll update everything to be nameof() in a future version

File: src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs (hunk 7 above)
Comment (COsborn2) [reply src/Chapter06/Listing06.20.ProvidingPropertyValidation.cs:38]: Updated to match book's text. I'll update everything to be nameof() in a future version

**Pull request closed without merging**